### PR TITLE
Update ASG module to v8.x

### PR DESCRIPTION
**Pull request by jeohist**
## Description of the change

> Update ASG module to 8.x. We are not affected by any of the breaking changes in 7.0 / 8.0, and it's nice to receive upstream updates.

## Type of change

- [X] New feature (non-breaking change that adds functionality);

### Development

- [X] All necessary variables have been defined, with defaults if applicable;
- [X] The code is formatted properly;

### Code review

- [ ] The module version is bumped accordingly;
- [ ] Spacelift tests are passing;
- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached;
- [ ] This pull request is no longer marked as "draft";
- [ ] Reviewers have been assigned;
- [ ] Changes have been reviewed by at least one other engineer;

### CODE REVIEW

File: iam.tf [hunk 1]
```diff
@@ -122,11 +122,14 @@ resource "aws_iam_role" "autoscaler" {
...
-  inline_policy {
-    name   = "ec2-autoscaler-${var.worker_pool_id}"
-    policy = data.aws_iam_policy_document.autoscaler[count.index].json
-  }
-
   depends_on = [module.asg]
   tags       = var.additional_tags
 }
+
+resource "aws_iam_role_policy" "autoscaler" {
```
Comment (lorengordon) [inline iam.tf:129]: Pretty sure this change is actually breaking for anyone that has an earlier version deployed. The user will need to write an `import` block in order to make it non-breaking. See the discussion on the provider repo: https://github.com/hashicorp/terraform-provider-aws/issues/39376#issuecomment-2363922272

Comment (adamconnelly) [reply iam.tf:129]: Just to mention as discussed in https://github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2/pull/103#issuecomment-2465023369, it looks like this doesn't create a backwards-compatibility issue. This PR just needs to be rebased against `main` now.

Comment (jeohist) [reply iam.tf]: Rebased :)
(sorry, don't check GitHub too often so I missed this)

Comment (adamconnelly) [reply iam.tf]: Thanks a lot @jeohist.

Sorry to be a pain, but could I ask you to do one more thing - update the version number [here](https://github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2/blob/5b0e2051294b532bd5c58951c39f489ed179e09e/.spacelift/config.yml#L2) to `2.6.4`? It just means we can get the module tests to run to sanity check the changes look good.

Comment (jeohist) [reply iam.tf]: Sure, done!

**Comment by jeohist** [comment]
Closing in favour of #105

**Comment by adamconnelly** [comment]
> Closing in favour of #105

Thanks @jeohist - I was just about to post a comment and explain. 🙇

**Pull request closed without merging**